tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -294,6 +334,85 @@ pub fn logic_layer_aggregation(
                 None => return Err(format_err!("No dataframes were returned."))
             };
 
+            let mut exclude_row_indexes: HashSet<i32> = HashSet::new();
+            let mut col_data_map: HashMap<usize, Vec<String>> = HashMap::new();
+
+            let mut unique_to_general_name_map: HashMap<String, String> = HashMap::new();
+
+            for (k, v) in unique_header_map.iter() {
+                let name: Vec<String> = k.split(".").map(|s| s.to_string()).collect();
+                let name_len = name.len();
+                let name = &name[name_len - 1];
+
+                unique_to_general_name_map.insert(
+                    format!("{} ID", v), format!("{} ID", name)
+                );
+            }
+
+            // This first pass will combine the data from the different dataframes.
+            // We also find the rows that will be ignored in the next pass.
+            for col_i in 0..num_cols {
+                let mut col_data: Vec<String> = vec![];
+
+                for df in &dfs {
+                    let c: &Column = &df.columns[col_i];
+                    let rows = c.stringify_column_data();
+                    col_data = [&col_data[..], &rows[..]].concat()
+                }
+
+                // Find rows that need to be excluded
```

Comment:
would it make more sense to have the exclusion happen in the query itself rather than return the rows and then filter them out?